src/lib/npmCommand.ts

```diff
@@ -160,11 +156,73 @@ export class NpmModule {
   }
 
   public show(registry: string): NpmShowResults {
-    return NpmCommand.runNpmCmd(`show ${this.module}@${this.version}`, { registry, cliRoot: this.cliRoot });
+    const showCmd = NpmCommand.runNpmCmd(`show ${this.module}@${this.version} --json`, {
+      registry,
+      cliRoot: this.cliRoot,
+    });
+
+    // `npm show` doesn't return exit code 1 when it fails to get a specific package version
+    // If `stdout` is empty then no info was found in the registry.
+    if (showCmd.stdout === '') {
+      throw new SfdxError(`Failed to find ${this.module}@${this.version} in the registry`, 'NpmError');
+    }
+
+    try {
+      return JSON.parse(showCmd.stdout) as NpmShowResults;
+    } catch (error) {
+      throw new SfdxError(error, 'ShellParseError');
+    }
   }
 
   public pack(registry: string, options?: shelljs.ExecOptions): void {
-    NpmCommand.runNpmCmd(`pack ${this.module}@${this.version}`, { ...options, registry, cliRoot: this.cliRoot });
+    try {
+      NpmCommand.runNpmCmd(`pack ${this.module}@${this.version}`, {
+        ...options,
+        registry,
+        cliRoot: this.cliRoot,
+      });
+    } catch (err) {
+      const sfdxErr = SfdxError.wrap(err);
+      throw new SfdxError(`Failed to fetch tarball from the registry: \n${sfdxErr.message}`, 'NpmError');
+    }
     return;
   }
+
+  public async fetchTarbarll(registry: string, options?: shelljs.ExecOptions): Promise<void> {
+    await this.pollForAvailability(() => {
+      this.pack(registry, options);
+    });
+    this.pack(registry, options);
```

Comment:
@cristiand391 Why are you calling `this.pack` twice in this function? 

Also minor thing, the function name has "Tarball" misspelled

Reply:
Since `fetchTarball` is called in `src/lib/installationVerification.ts`, every time I ran the tests the polling logic blocked the execution for minutes so to avoid that I moved the polling logic to `pollForAvailability` and called it in `fetchTarball`, that way it can be easily stubbed in tests to always resolve the promise.
The second call to `this.pack` should never be stubbed so it has to be called again after polling for tests to properly work.